pkg/assetvalidator/validateAssets.go

```diff
@@ -9,22 +9,42 @@ import (
 	"github.com/asecurityteam/nexpose-asset-producer/pkg/domain"
 )
 
+// NexposeScanType is a type alias to signify type of scan according to Nexpose
+type NexposeScanType string
+
+const (
+	automated NexposeScanType = "Automated"
+	manual    NexposeScanType = "Manual"
+	scheduled NexposeScanType = "Scheduled"
+	agent     NexposeScanType = "Agent"
+)
+
+// ScanType is a type alias that generifies a scan to either local or remote
+// The purpose is to not expose Nexpose specifics further down the pipeline
+type ScanType string
+
+const (
+	local   ScanType = "local"
+	remote  ScanType = "remote"
+	unknown ScanType = "unknown"
+)
+
 // NexposeAssetValidator is used to validate a list of retrieved Assets
 type NexposeAssetValidator struct {
 }
 
 // ValidateAssets takes in a list of assets from Nexpose, and returns a list of valid Asset events, and a list of errors based
 // on defined validation rules
-func (v *NexposeAssetValidator) ValidateAssets(ctx context.Context, assets []domain.Asset, scanID string) ([]domain.AssetEvent, []error) {
+func (v *NexposeAssetValidator) ValidateAssets(ctx context.Context, assets []domain.Asset, scanInfo domain.ScanInfo) ([]domain.AssetEvent, []error) {
 	assetEventListList := []domain.AssetEvent{}
 	errorList := []error{}
 	for _, asset := range assets {
-		scanTime, err := v.getScanTime(asset, scanID)
+		scanTime, scanType, err := v.getScanTimeAndScanType(asset, scanInfo)
 		if err != nil {
 			errorList = append(errorList, err)
 			continue
 		}
-		assetEvent, err := v.assetPayloadToAssetEvent(asset, scanTime)
+		assetEvent, err := v.assetPayloadToAssetEvent(asset, scanTime, string(scanType))
```

Comment:
task: since you've made `NexposeScanType` and `ScanType` custom types to indicate that they should only ever be one of the acceptable values (which I like!), then you could also implement the `Stringer` interface for each type to not have to do these type conversions anywhere in the code: https://tour.golang.org/methods/17.

Reply:
done!